pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -29,6 +29,8 @@
 
 
 public class CommonConstants {
+  public static final String GROOVY_STATIC_ANALYZER_CONFIG = "pinot.server.groovy.static.analyzer";
```

Comment:
This shouldn't be added as top level config. We need this config for both controller and broker, and they can potentially be different (one is for ingestion, and one is for query). Take a look at the subclass `Controller` and `Broker` within this class.
We can also consider adding a common one to be used when there is no controller/broker specific config

Reply:
I will use service name only and not use case specific name like `pinot.broker.*` and 'pinot.controller.*` to continue the existing pattern.

Reply:
With this, I think, we can avoid the specific REST APIs too for the groovy config and use `POST /cluster/configs` API to configure the groovy static analyzer.

Reply:
Sounds good. Since we are not keeping the instance type prefix, we can put them in the top level, or add a subclass `Groovy`. Let's not put it in the beginning of the file, and also add some documentation for them.

Regarding the REST API, the ser/de of this config is not easy, so I'd suggest keeping the REST API, but add a type param which can be `ingestion`, `query` or `all`. Same for the GET request, where we perform the same fallback logic based on the type asked